src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -592,7 +592,7 @@ ONNXEntryPointOp ONNXEntryPointOp::create(mlir::Location location,
 /// Infer the output shape of the ONNXExpOp. This method is required by the
 /// shape inference interface.
 LogicalResult ONNXExpOp::inferShapes(
-    std::function<void(mlir::FuncOp)> shapeInferenceFunc) {
+    std::function<void(mlir::Region &)> shapeInferenceFunc) {
```

Comment:
minor: maybe rename parameter to shapeInferenceRegion

Reply:
Maybe doShapeInference? The Func suffix is there to suggests it's something callable.

Reply:
sure